custom_components/unifiprotect/select.py

```diff
@@ -207,27 +238,46 @@ class UnifiProtectSelects(UnifiProtectEntity, SelectEntity):
 
     def __init__(
         self,
-        upv_object: UpvServer,
+        protect: ProtectApiClient,
         protect_data: UnifiProtectData,
-        server_info: dict[str, Any],
-        device_id: int,
+        device: ProtectAdoptableDeviceModel,
         description: UnifiProtectSelectEntityDescription,
         options: list[dict[str, Any]],
     ):
         """Initialize the unifi protect select entity."""
-        super().__init__(upv_object, protect_data, server_info, device_id, description)
-        self._attr_name = f"{self.entity_description.name} {self._device_data['name']}"
+        super().__init__(protect, protect_data, device, description)
+        self.device: Camera | Viewer | Light = device
+        self._attr_name = f"{self.device.name} {self.entity_description.name}"
         self._attr_options = [item["name"] for item in options]
-        self._data_key = description.ufp_device_data_key
+        self._data_key = description.ufp_value
         self._hass_to_unifi_options = {item["name"]: item["id"] for item in options}
         self._unifi_to_hass_options = {item["id"]: item["name"] for item in options}
 
     @property
     def current_option(self) -> str:
         """Return the current selected option."""
-        unifi_value = self._device_data[self._data_key]
-        if self.entity_description.key == _KEY_DOORBELL_TEXT:
-            return unifi_value
+        unifi_value = get_nested_attr(self.device, self._data_key)
+
+        if isinstance(unifi_value, Enum):
+            unifi_value = unifi_value.value
+
+        if isinstance(unifi_value, Liveview):
+            unifi_value = unifi_value.id
```

Comment:
Can this be an `if/elif` ?

Reply:
That was on purpose. Originally. An earlier iteration of this had it so the value could be an Enum and still hit one of the follow ifs. Does not look like it is needed now though.